### PR TITLE
square component

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: client/src/components/board.rs [hunk 1]
```diff
@@ -19,6 +12,14 @@ use dioxus::prelude::*;
 use futures::executor::block_on;
 use once_cell::sync::Lazy;
 
+use crate::arrows::{ArrowData, Arrows};
```
Comment (DanielAlbl) [inline client/src/components/board.rs:15]: Did rustrover like the `use crates` later?

Comment (Dillon-Roller) [reply client/src/components/board.rs:15]: no, but i felt like having the crate stuff below other dependencies were better, just visually grouping our source code closer to the uses of our source code

File: client/src/components/square.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    perspective: Color,
+) -> Element {
+    let top_left = to_point(pos, *board_size, *perspective);
+    cx.render(rsx! {
+        div {
+            class: "{class}",
+            style: "
+                left: {top_left.x}px;
+                top: {top_left.y}px;
+                width: {cx.props.board_size / 8}px;
```
[the remark this answers is not in the document]
Comment (Dillon-Roller) [reply client/src/components/square.rs:22]: yeah we can do that

File: client/src/components/board.rs [hunk 3]
```diff
@@ -327,16 +324,12 @@ pub(crate) fn Board(cx: Scope<BoardProps>) -> Element {
             if !game.read().is_replaying() && selected_piece.read().is_some() {
```
[the remark this answers is not in the document]
Comment (Dillon-Roller) [reply client/src/components/board.rs:324]: I'm honestly not sure

**Review by DanielAlbl — DISMISSED** [review]

File: client/src/components/board.rs [hunk 4]
```diff
@@ -296,52 +297,43 @@ pub(crate) fn Board(cx: Scope<BoardProps>) -> Element {
...
-                        ",
+                    BoardSquare {
+                        class: class,
+                        pos: pos,
+                        board_size: cx.props.size,
+                        perspective: cx.props.perspective,
                     }
                 }
             }),
+            if !game.read().is_replaying() && selected_piece.read().is_some() {
```
Comment (DanielAlbl) [inline client/src/components/board.rs:309]: Can we do `if let` here?

**Review by DanielAlbl — APPROVED** [review]

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**